packages/wrangler/src/config/environment.ts

```diff
@@ -0,0 +1,305 @@
+   * have a .js file be treated as an ESModule instead of CommonJS.
+   *
+   * @inheritable
+   */
+  rules: Rule[];
+
+  /**
+   * TODO: remove this as it has been deprecated.
+   *
+   * This is just here for now because the `route` commands use it.
+   * So we need to include it in this type so it is available.
+   */
+  zone_id?: string;
+}
+
+/**
+ * The `EnvironmentNonInheritable` interface declares all the configuration fields for an environment
+ * that cannot be inherited from the top-level environment, and must be defined specifically.
+ *
+ * If any of these fields are defined at the top-level then they should also be specifically defined
+ * for each named environment.
+ */
+interface EnvironmentNonInheritable {
+  /**
+   * A map of environment variables to set when deploying your worker.
+   *
+   * NOTE: This field is not automatically inherited from the top level environment,
+   * and so must be specified in every named environment.
+   *
+   * @default `{}`
+   * @nonInheritable
+   */
+  vars: { [key: string]: unknown };
+
+  /**
+   * A list of durable objects that your worker should be bound to.
+   *
+   * For more information about Durable Objects, see the documentation at
+   * https://developers.cloudflare.com/workers/learning/using-durable-objects
+   *
+   * NOTE: This field is not automatically inherited from the top level environment,
+   * and so must be specified in every named environment.
+   *
+   * @default `{bindings:[]}`
+   * @nonInheritable
+   */
+  durable_objects: {
+    bindings: {
+      /** The name of the binding used to refer to the Durable Object */
+      name: string;
+      /** The exported class name of the Durable Object */
+      class_name: string;
+      /** The script where the Durable Object is defined (if it's external to this worker) */
+      script_name?: string;
+    }[];
+  };
+
+  /**
+   * These specify any Workers KV Namespaces you want to
+   * access from inside your Worker.
+   *
+   * To learn more about KV Namespaces,
+   * see the documentation at https://developers.cloudflare.com/workers/learning/how-kv-works
+   *
+   * NOTE: This field is not automatically inherited from the top level environment,
+   * and so must be specified in every named environment.
+   *
+   * @default `[]`
+   * @nonInheritable
+   */
+  kv_namespaces: {
+    /** The binding name used to refer to the KV Namespace */
+    binding: string;
+    /** The ID of the KV namespace */
+    id: string;
+    /** The ID of the KV namespace used during `wrangler dev` */
+    preview_id?: string;
+  }[];
+
+  /**
+   * Specifies R2 buckets that are bound to this Worker environment.
+   *
+   * NOTE: This field is not automatically inherited from the top level environment,
+   * and so must be specified in every named environment.
+   *
+   * @default `[]`
+   * @nonInheritable
+   */
+  r2_buckets: {
+    /** The binding name used to refer to the R2 bucket in the worker. */
+    binding: string;
```

Comment:
I think this should be `name`?

Reply:
LOL! Sunil said that too.
This is currently what Wrangler1 does. So we are copying that here.
Happy to have a discussion about this moving forward.

Reply:
Note that `kv_namespaces` also use `binding`.

Reply:
But durable_objects use `name` 🤷

Reply:
Maybe we should move them all to `name`, with `binding` as deprecated? /cc @threepointone

Reply:
Yeah we should definitely do a sweep of all of these. Removing the unnecessary `bindings` key on durable, etc etc. I'm wary of doing the deprecation right now, this all would be so much simpler when we have auto toml editing.